src/System.Data.SqlClient/src/System/Data/SqlClient/SqlTransaction.cs

```diff
@@ -141,47 +160,94 @@ override public void Commit()
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing)
+            Exception e = null;
```

Comment:
@saurabh500 To trace a case when a transaction disposed without explicit rollback / commit.